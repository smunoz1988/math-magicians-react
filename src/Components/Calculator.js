import React, { useState } from 'react';
//  import calculate from '../logic/calculate';

const Calculator = () => {
  const buttons = ['AC', '+/-', '%', '/', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];

  const [count, setCount] = useState(0);

  function handleClick() {
    setCount(count + 1);
  }

  return (
    <>
      <div className="calculator">
        <p className="outputResult">{count}</p>
        <div className="buttonContainer">
          {buttons.map((button) => (
            <button type="button" className="calcButtons" key={button} onClick={handleClick}>{button}</button>
          ))}
        </div>
      </div>
    </>
  );
};

export default Calculator;
