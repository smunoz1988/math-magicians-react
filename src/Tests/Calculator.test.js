import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });
});