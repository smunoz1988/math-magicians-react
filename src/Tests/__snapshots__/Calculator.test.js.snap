// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculatorSection"
  >
    <h3>
      Lets do some math!
    </h3>
    <div
      class="calculator"
    >
      <p
        class="outputResult"
      >
        0
      </p>
      <div
        class="buttonContainer"
      >
        <button
          class="calcButtons"
          type="button"
        >
          AC
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          +/-
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          %
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          รท
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          7
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          8
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          9
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          x
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          4
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          5
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          6
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          -
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          1
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          2
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          3
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          +
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          0
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          .
        </button>
        <button
          class="calcButtons"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
